loyalty_multi_gift/views/coupon_program_views.xml

```diff
@@ -1,32 +1,14 @@
 <?xml version="1.0" encoding="utf-8" ?>
 <odoo>
-    <record id="coupon_program_view_form_common" model="ir.ui.view">
-        <field name="inherit_id" ref="coupon.coupon_program_view_form_common" />
-        <field name="model">coupon.program</field>
+    <record id="loyalty_program_view_form_common" model="ir.ui.view">
+        <field name="inherit_id" ref="loyalty.loyalty_program_view_form" />
+        <field name="model">loyalty.program</field>
         <field name="arch" type="xml">
-            <xpath expr="//field[@name='reward_product_id']/.." position="attributes">
-                <attribute
-                    name="attrs"
-                >{'invisible': [('reward_type', '=', 'multi_gift')]}</attribute>
-            </xpath>
-            <xpath expr="//field[@name='reward_product_id']/.." position="before">
-                <group attrs="{'invisible': [('reward_type', '!=', 'multi_gift')]}">
-                    <field name="coupon_multi_gift_ids">
-                        <tree editable="top">
-                            <field name="reward_product_quantity" string="Qty" />
-                            <field
-                                name="reward_default_product_id"
-                                string="Default gift"
-                                readonly="1"
-                            />
-                            <field
-                                name="reward_product_ids"
-                                widget="many2many_tags"
-                                required="1"
-                            />
-                        </tree>
-                    </field>
-                </group>
+            <xpath
+                expr="//form/sheet/group/group/field[@name='available_on']"
+                position="after"
+            >
+                <field name="get_all_rewards" />
```

Comment:
For this, I'd add an special program type and use the method that sets the default values for that case.